custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/java_predictor.py

```diff
@@ -150,10 +150,20 @@ def has_read_input_data_hook(self):
         return False
 
     def predict(self, **kwargs):
-        input_text_data = kwargs.get(StructuredDtoKeys.BINARY_DATA).decode("utf-8")
+        input_text_bytes = kwargs.get(StructuredDtoKeys.BINARY_DATA)
 
         start_predict = time.time()
-        out_csv = self._predictor_via_py4j.predict(input_text_data)
+        # If data size is more than 33K, pass it as a file to Java,
+        # as passing big chunks to py4j as an array is 10% slower
+        DATA_BUFFER_LIMIT_33K = 33792
+        if len(input_text_bytes) > DATA_BUFFER_LIMIT_33K:
+            with tempfile.NamedTemporaryFile(mode="wb") as tf:
```

Comment:
Have you considered using a memory mapped file? This may increase performance in a great deal.

Reply:
created a ticket